pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -337,6 +337,46 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(env model.Env
 	return listeners
 }
 
+type listenerEntry struct {
+	service *model.Service
+	servicePort *model.Port
+	listener *xdsapi.Listener
+}
+
+func getServiceName(env model.Environment, service *model.Service) string {
+	attrs, err := env.GetServiceAttributes(service.Hostname)
+	if err != nil || attrs == nil || attrs.Name == "" {
+		return "unknown"
+	}
+	return fmt.Sprintf("%s/%s", attrs.Namespace, attrs.Name)
+}
+
+func getClusterName(service *model.Service, servicePort *model.Port) string {
+	return model.BuildSubsetKey(model.TrafficDirectionOutbound, "",
+		service.Hostname, servicePort.Port)
+}
```

Comment:
This also seems like an unnecessary function just for the printing convenience. Given that its just two more args, and called/used only once in this entire file.

Reply:
The function simplifies the call site